neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py

```diff
@@ -1654,10 +1654,23 @@ def process_network_ports(self, port_info, ovs_restarted):
         self._add_port_tag_info(need_binding_devices)
         if security_disabled_ports:
             added_ports -= set(security_disabled_ports)
-        self.sg_agent.setup_port_filters(added_ports,
-                                         port_info.get('updated', set()))
+
+        # The following call was split into two calls so that added_ports
+        # can be bound quicker.
+        #self.sg_agent.setup_port_filters(added_ports,
+        #                                 port_info.get('updated', set()))
+
+        devices_to_refilter, global_refresh_firewall = (
+            self.sg_agent.setup_port_filters_part_1(added_ports))
+
         failed_devices['added'] |= self._bind_devices(need_binding_devices)
```

Comment:
Looking at your scenario, it is my understanding that _bind_devices()  method will only be called once for that OVS port.  It will be called when that OVS port is allocated to that compute node and placed on that specific private network.   This will occur before the first CLI command you list.

The admin_state  and security group changes (via the CLI commands) on an existing port should not ever invoke the _bind_devices()  method a 2nd time.... but instead will be updated when the  refresh_firewall()  operation is done.

The change that I have in this pull request should have no impact on the CLI operations that you list above.